share/longtail/climbing/crawler.pl

```diff
@@ -0,0 +1,147 @@
+	my $type = $tree->findvalue('//a[@href="/article/AreaTypes"]');
+	
+	unless($type eq "region"){
+		$data->{$place}->{"url"}=$url;
+		print "$place: $url \n";
+		$data->{$place}->{"type"}=$type;
+		
+		my @latmeta = $tree->findnodes('//meta[@property="place:location:latitude"]');
+		my @lonmeta = $tree->findnodes('//meta[@property="place:location:longitude"]');
+		my $lat = $latmeta[0]->attr('content');
+		my $lon = $lonmeta[0]->attr('content');
+		$data->{$place}->{"longitude"} = $lon;
+		$data->{$place}->{"latitude"} = $lat;
+		
+		my @breadcrumbs = $tree->findvalues('//div[@id="breadCrumbs"]/ul/li');
+		foreach my $i(1..$#breadcrumbs){
+			$data->{$place}->{"breadcrumbs"}->[$i-1] = $breadcrumbs[$i];
+			}
+
+		my $numroutes=$tree->findvalue('//a[@title="Search and filter these routes"]');
+		$numroutes = substr($numroutes,1,(index($numroutes,'routes')-2));
+		$data->{$place}->{"number of routes"}=$numroutes;
+
+		my @styles = $tree->findvalues('//span[@class= "style-breakdown" ]/a');
+		foreach my $i(0..$#styles){
+			my $style = $styles[$i];
+			my $percent = substr($style,1,index($style,"%"));
+			my $type = substr ($style,index($style,"%")+1);
+			$type =~ s/^\s+//g;
+			$data->{$place}->{'styles'}->{$type} = $percent; 
+			}
+
+		my @areas = $tree->findnodes('//div[@class = "area"]' );
+		foreach my $i (0..$#areas){
+			$data->{$place}->{'areas'}->[$i]->{'name'} = substr($areas[$i]->findvalue('div[@class="name"]'),0,-5);
+			$data->{$place}->{'areas'}->[$i]->{'routes'} = $areas[$i]->findvalue('div/div[@class="routes"]');
+			$data->{$place}->{'areas'}->[$i]->{'ticks'} = $areas[$i]->findvalue('div/div[@class="ticks"]');
+			$data->{$place}->{'areas'}->[$i]->{'height'} = $areas[$i]->findvalue('div/div[@class="height"]');
+			my @links= $areas[$i]->findnodes('div/a[@href]');
+			$data->{$place}->{'areas'}->[$i]->{'link'} = $links[1]->attr('href');
+		}
+		
+		my $paragraph = "";
+		my @summaryinfo= $tree->findnodes('//div[@class="node-info summary"]');
+		my @descriptioninfo= $tree->findnodes('//div[@class="node-info description"]');
+		if($#summaryinfo == 0){
+			my @summary= $summaryinfo[0]->findvalues('div/div/p');
+			for my $i (0 .. $#summary){
+				$paragraph = ($summary[$i] . "\n\n");
+			}
+		}
+		if($#descriptioninfo == 0){
+			my @description = $descriptioninfo[0]->findvalues('div/div/p');
+			my $paragraphs = $#description;
+			if( $paragraphs > 1){
+				$paragraphs = 1;
+			}
+			for my $i (0 .. $paragraphs){
+				my  $end = "";
+				unless($i == $paragraphs) {$end = "\n\n"}
+				$paragraph .= ($description[$i] . $end);
+			}
+		}
+		if(($paragraph eq "") == 1)
+		{
+			my @defmeta = $tree->findnodes('//meta[@property="og:description"]');
+			my $definition = $defmeta[0]->attr('content');
+			$definition = substr($definition, index($definition, $place)+length($place)+2);
+			$paragraph = $definition;
+		}
+		
+		
+		
+		
+		print OUT <<EOH
+<doc>
+     <field name="title"><![CDATA[$place]]></field>
```

Comment:
Could you append the word "Climbing" to all the titles?  That's going to help us a lot with the relevancy.